lua/lspconfig/configs.lua

```diff
@@ -56,13 +54,31 @@ function configs.__newindex(t, config_name, config_def)
 
     local config = tbl_deep_extend('keep', user_config, default_config)
 
+    local get_augroup_id = function()
+      local group_name = 'lspconfig_' .. config.name
+      local group_id
+      if not configs.lsp_groups[group_name] then
+        group_id = api.nvim_create_augroup(group_name, { clear = true })
+        configs.lsp_groups[group_name] = group_id
+      else
+        group_id = configs.lsp_groups[group_name]
+      end
+      return group_id
+    end
+
+    if not configs.lsp_groups then
+      configs.lsp_groups = {}
```

Comment:
This is a red flag. We should not need a cache or a registry for this.

Reply:
There's `on_exit` which should be a good fit to use for this purpose.

- use a unique name (some kind of hash?) for the augroup
- or: pass `create_augroup` to every instance of `create_autocmd`

```lua
on_exit = function()
  vim.schedule(function()
    pcall(function()
      vim.api.nvim_clear_autocmds { group = name }
      -- or 
      -- vim.api.nvim_del_augroup_by_name(name)
    end)
  end)
end
```